From: Simon Vogl <simon.vogl@gmx.net>
Date: Mon, 17 Jul 2020 21:15:29 UTC
Subject: [PATCH] Fix broken python detection of QMPlay2 when it tries to launch youtube-dl

This patch is required to prevent QMPlay2 from always preferring python2 over python3 which can lead to crashes in certain cases.

--- a/src/qmplay2/YouTubeDL.cpp
+++ b/src/qmplay2/YouTubeDL.cpp
@@ -459,25 +459,9 @@
         const int idx = shebang.lastIndexOf("python");
         if (shebang.startsWith("#!") && idx > -1)
         {
-            const auto pythonCmd = shebang.mid(idx);
-            if (!QStandardPaths::findExecutable(pythonCmd).endsWith(pythonCmd))
-            {
-                QStringList pythonCmdsToCheck {
-                    "python",
-                    "python2",
-                    "python3",
-                };
-                pythonCmdsToCheck.removeOne(pythonCmd);
-                for (auto &&pythonCmd : asConst(pythonCmdsToCheck))
-                {
-                    if (QStandardPaths::findExecutable(pythonCmd).endsWith(pythonCmd))
-                    {
-                        args.prepend(program);
-                        program = pythonCmd;
-                        break;
-                    }
-                }
-            }
+            const auto pythonCmd = "python3";
+            args.prepend(program);
+            program = pythonCmd;
         }
         ytDlFile.close();
     }
